massa-protocol-worker/src/protocol_worker.rs

```diff
@@ -218,79 +233,63 @@ impl ProtocolWorker {
     /// - `node_event_rx`
     /// And at the end every thing is closed properly
     /// Consensus work is managed here.
-    /// It's mostly a `tokio::select!` within a loop.
-    pub async fn run_loop(mut self) -> Result<NetworkEventReceiver, ProtocolError> {
-        // TODO: Config variable for the moment 10000 (prune) (100 seconds)
-        let operation_prune_timer = sleep(self.config.asked_operations_pruning_period.into());
-        tokio::pin!(operation_prune_timer);
-        let block_ask_timer = sleep(self.config.ask_block_timeout.into());
-        tokio::pin!(block_ask_timer);
-        let operation_batch_proc_period_timer =
-            sleep(self.config.operation_batch_proc_period.into());
-        tokio::pin!(operation_batch_proc_period_timer);
-        let operation_announcement_interval =
-            sleep(self.config.operation_announcement_interval.into());
-        tokio::pin!(operation_announcement_interval);
+    /// It's mostly a `crossbeam::select!` within a loop.
+    pub fn run_loop(mut self) -> Result<NetworkEventReceiver, ProtocolError> {
         loop {
             massa_trace!("protocol.protocol_worker.run_loop.begin", {});
-            /*
-                select! without the "biased" modifier will randomly select the 1st branch to check,
-                then will check the next ones in the order they are written.
-                We choose this order:
-                    * manager commands: low freq, avoid having to wait to stop
-                    * incoming commands (high frequency): process commands in priority (this is a high-level crate so we prioritize this side to avoid slowing down consensus)
-                    * network events (high frequency): process incoming events
-                    * ask for blocks (timing not important)
-            */
-            tokio::select! {
+            select! {
```

Comment:
Why we still use a select pattern ? The goal wasn't to have threads that manage separately each workflow ?

Reply:
> to have threads that manage separately each workflow

I think this could be reworded as "manage each **separate** workflow". Meaning that there is only a point in doing something like that if the workflow is actually separate--which would be indicated by not having to share state with other workflows(other then some sort of communication device). 

In the meantime the workers still need to communicate with various other workers, and that is best done with a select statement over multiple channels. 

You could have separate threads handle each channel, but then you'd have to share state between those threads and they would likely block each other with locks. So having one thread do a select is the equivalent of having multiple threads block each other with locks.